packages/lodestar/src/db/api/beacon/repositories/blockArchive.ts

```diff
@@ -18,14 +23,36 @@ export class BlockArchiveRepository extends Repository<Slot, SignedBeaconBlock>
   }
 
   public async put(key: Slot, value: SignedBeaconBlock): Promise<void> {
-    await Promise.all([super.put(key, value), this.storeRootIndex(value), this.storeParentRootIndex(value)]);
+    const blockRoot = this.config.types.BeaconBlock.hashTreeRoot(value.message);
+    const slot = value.message.slot;
+    await Promise.all([
+      super.put(key, value),
+      this.storeRootIndex(slot, blockRoot),
+      this.storeParentRootIndex(slot, value.message.parentRoot),
+    ]);
   }
 
   public async batchPut(items: ArrayLike<IKeyValue<Slot, SignedBeaconBlock>>): Promise<void> {
     await Promise.all([
       super.batchPut(items),
-      Array.from(items).map((item) => this.storeRootIndex(item.value)),
-      Array.from(items).map((item) => this.storeParentRootIndex(item.value)),
+      Array.from(items).map((item) => {
+        const slot = item.value.message.slot;
+        const blockRoot = this.config.types.BeaconBlock.hashTreeRoot(item.value.message);
+        return this.storeRootIndex(slot, blockRoot);
+      }),
+      Array.from(items).map((item) => {
+        const slot = item.value.message.slot;
+        const parentRoot = item.value.message.parentRoot;
+        return this.storeParentRootIndex(slot, parentRoot);
+      }),
+    ]);
+  }
+
+  public async batchPutBinary(items: ArrayLike<IKeyValueSummary<Slot, Buffer, IBlockSummary>>): Promise<void> {
```

Comment:
this is the one to be used in our `archiveBlocks` task later (block summaries are available in forkchoice)